skimage/exposure/_adapthist.py

```diff
@@ -244,7 +244,8 @@ def clip_histogram(hist, clip_limit):
     n_excess -= mid.size * clip_limit - mid.sum()
     hist[mid_mask] = clip_limit
 
-    while n_excess > 0:  # Redistribute remaining excess
+    iteration = 0
+    while n_excess > 0 and iteration < 20:  # Redistribute remaining excess
```

Comment:
How did you choose 20? I'm uncomfortable with it being hardcoded like this. At the very least, I would create a variable `max_iterations` and with it a comment about why it exists and how its value was chosen.